fighthealthinsurance/rest_views.py

```diff
@@ -1,15 +1,123 @@
-from asgiref.sync import async_to_sync
 import json
+import typing
+
+from asgiref.sync import async_to_sync
+from django.conf import settings
 
-from fighthealthinsurance.common_view_logic import *
-from fighthealthinsurance.rest_serializers import *
+from rest_framework import status
+from rest_framework import viewsets
 from rest_framework.response import Response
+from rest_framework.serializers import Serializer
 from rest_framework.views import APIView
-from rest_framework import status
+
+from fighthealthinsurance import common_view_logic
+from fighthealthinsurance.model_router import model_router
+from fighthealthinsurance import rest_serializers as serializers
 
 from stopit import ThreadingTimeout as Timeout
 
 
+class SerializerMixin:
+    serializer_class: typing.Optional[Serializer] = None
+
+    def get_serializer(self, data=None):
+        if self.serializer_class is None:
+            raise ValueError(
+                'serializer_class must be defined and not None'
+            )
+
+        return self.serializer_class(data=data)
+
+
+class CreateMixin(SerializerMixin):
+    def perform_create(self, request, serializer):
+        pass
+
+    def create(self, request):
+        request_serializer = self.get_serializer(data=request.data)
+        request_serializer.is_valid(raise_exception=True)
+        response_serializer = self.perform_create(request, request_serializer)
+
+        if response_serializer:
+            result = response_serializer.data
+        else:
+            result = None
+
+        return Response(result, status=status.HTTP_201_CREATED)
+
+
+class DeleteMixin(SerializerMixin):
+    def perform_delete(self, request, serializer):
+        pass
+
+    def delete(self, request, *args, **kwargs):
+        """ For some reason """
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        self.perform_delete(request, serializer, *args, **kwargs)
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class DeleteOnlyMixin:
```

Comment:
This is a hack that I made to get around delete-only viewsets not showing in the router. There's probably a better way, like defining actions specifically, but I didn't want to spend too much time on it since its not necessary right now.